### PR TITLE
Part from room when Faye says we left

### PR DESCRIPTION
**Pull request by Gargaj**
Remove the user from the IRC room when a "remove" operation with no user comes in from Faye

### CODE REVIEW

File: lib/gitter-adapter.js [hunk 1]
```diff
@@ -214,11 +214,20 @@ Adapter.prototype.subscribeToRoom = function(room) {
...
+      {
+        // someone else left
+        nick = evt.model.username;
+        mask = ':' + nick + '!' + nick + '@irc.gitter.im';
+        c.send(mask, 'PART', '#' + room.uri);
+      }
+      else
+      {
+        // we left
+        this.leaveRoom( '#' + room.uri);
```
Comment (MadLittleMods) [inline lib/gitter-adapter.js]: No need for the channel hash in front, we just strip it in the function anyway

File: lib/gitter-adapter.js [hunk 2]
```diff
@@ -214,11 +214,20 @@ Adapter.prototype.subscribeToRoom = function(room) {
...
-      nick = evt.model.username;
-      mask = ':' + nick + '!' + nick + '@irc.gitter.im';
-      c.send(mask, 'PART', '#' + room.uri);
+      if (evt.model.username)
+      {
+        // someone else left
+        nick = evt.model.username;
+        mask = ':' + nick + '!' + nick + '@irc.gitter.im';
+        c.send(mask, 'PART', '#' + room.uri);
+      }
```
Comment (MadLittleMods) [inline lib/gitter-adapter.js]: Update `if` statement to match brackets on same line style

File: lib/gitter-adapter.js (hunk 1 above)
Comment (Gargaj) [inline lib/gitter-adapter.js]: Check again:
```
Adapter.prototype.leaveRoom = function(channel) {
  var uri = channel.replace('#', '').toLowerCase();
  this.client.send(this.client.mask(), 'PART', channel);
  this._unsubscribe(uri);
};
```
You're talking about ```uri```, not ```channel```.

Comment (MadLittleMods) [inline lib/gitter-adapter.js]: @Gargaj Good spot 😉 for the `PART` message

**Comment by MadLittleMods** [comment]
Thanks @Gargaj 😀

**Pull request merged**